lib/erb_lint/linters/deprecated_classes.rb

```diff
@@ -22,18 +22,14 @@ def initialize(config)
         @addendum = config.fetch('addendum', '')
       end
 
-      protected
-
-      def lint_lines(lines)
+      def lint_file(file_tree)
         errors = []
 
-        lines.each_with_index do |line, index|
-          start_tags = StartTagHelper.start_tags(line)
-          start_tags.each do |start_tag|
-            start_tag.attributes.select(&:class?).each do |class_attr|
-              class_attr.value.split(' ').each do |class_name|
-                errors.push(*generate_errors(class_name, index + 1))
-              end
+        html_elements = Parser.filter_erb_nodes(file_tree)
+        html_elements.each do |html_element|
+          html_element.attribute_nodes.select { |attribute| attribute.name.casecmp('class') == 0 }.each do |class_attr|
```

Comment:
These lines are a little hard to read.

Could you please break this out into some well-named variables? Something like:

```
html_element_class_attributes = html_element.attribute_nodes.select { |attribute| attribute.name.casecmp('class') == 0 }

html_element_class_attributes.each do |class_attr|
```